File: frontend/src/app/shared/pagination.ts
export interface Pagination<T, P> {
    content: T[];
    totalElements: number;
    numberOfElements: number;
    size: number;
    number: number;
    currentPage: number;
    lastPage: number;
    nextPage: number;
    prevPage: number;
    links: {
        next: string;
        prev: string;
        last: string;
        first: string;
    }
    params: P;
}

File: pacientes/src/main/java/br/com/basis/madre/domain/Pagination.java
package br.com.basis.madre.domain;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Pagination<T, P> {

    private T content;
    private Integer currentPage;
    private Integer nextPage;
    private Integer prevPage;
    private Integer lastPage;
    private Long numberOfElements;
    private PaginationLinks links;
    private P params;

}
